File: longest_substring_without_repeating_characters.java
// https://leetcode.com/problems/longest-substring-without-repeating-characters/description/

class Solution {
  public int lengthOfLongestSubstring(String s) {
    int res=0;
    for (int i=0;i<s.length();i++){
      int len=0;
      Set<Character> set=new HashSet<Character>();
      for(int j=i;j<s.length();j++){
        if(set.contains(s.charAt(j))){
          break;
        }
        set.add(s.charAt(j));
        len++;
      }
      if (res<len){
        res=len;
      }
    }
    return res;
  }
}

File: longest-substring-without-repeating-characters.go

//https://leetcode.com/problems/longest-substring-without-repeating-characters/description/

func lengthOfLongestSubstring(s string) int {
    res:=0
    for i:=0; i<len(s); i++ {
        length:= 0
        set:= make(map[byte]bool)
        for j := i; j < len(s); j++ {
            if set[s[j]] == true {
                break
            }
            set[s[j]]=true
            length++
        }
        if res<length{
            res=length
        }
    }
    return res
}
